src/google/protobuf/stubs/common.h

```diff
@@ -47,6 +47,10 @@
 #include "google/protobuf/stubs/platform_macros.h"
 #include "google/protobuf/stubs/port.h"
 
+#if !defined(PROTOBUF_FORCE_OLD_CPP_VERSION_WE_KNOW_WHAT_WE_ARE_DOING) && __cplusplus < 201402L
```

Comment:
This is causing failures in our windows and aarch64 builds.  I suspect you may need some variant of __cplusplus for these platforms

Reply:
See above, I like the guard a lot.  It's a good idea, I just think we need to either drop it on certain platforms or use alternatives

Reply:
out of curiosity, what compiler is the aarch64 build using that it doesn't have __cplusplus?

Reply:
That's a good question.  IIRC our aarch64 are using an ancient version of gcc that's actually well outside our support window.  I have a TODO to upgrade those, but it's non-trivial